### ISSUE TITLE
Suggested fix for `no-div-regex` trips `no-useless-escape`

### ISSUE THREAD

**Issue by coreyfarrell** [eslint/eslint#11355]
**Tell us about your environment**

* **ESLint Version:** 5.13
* **Node Version:** 10.15.0
* **npm Version:** 6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
	"rules": {
		"no-div-regex": "error",
		"no-useless-escape": "error"
	}
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
console.log('test=='.replace(/=*$/, '')); // no-div-regex error
console.log('test=='.replace(/\=*$/, '')); // no-useless-escape error
console.log('test=='.replace(/[=]*$/, '')); // accepted by both rules
```

```bash
eslint script.js
```

**What did you expect to happen?**
`no-div-regex` should suggest replacing `/=foo/` with `/[=]foo/` to avoid `no-useless-escape` error.  Alternatively `no-useless-escape` could accept an `ignoreDivRegex` boolean option to allow `/\=foo/`.

**Are you willing to submit a pull request to fix this bug?**
I'm sure I can handle the documentation approach of suggesting `/[=]foo/`.  I'm not confident I can create the `no-useless-escape` option to ignore div regex.

**Comment by xianghongai** [comment]
`no-useless-escape`

```javascript
name = name.replace(/[\[\]]/g, "\\$&");
```

**Comment by platinumazure** [comment]
Oops, this slipped through the cracks. Sorry about that.

I think this is a bug and I think the best option is to have `no-div-regex` use `[=]` in its autofix. I'll reopen this and mark as accepted.

@coreyfarrell Would you be willing to submit a PR? If not, no worries, just wanted to see if you were still interested!

**Comment by coreyfarrell** [comment]
@platinumazure thanks for responding.  Unfortunately I don't currently have time to work on this and since `xo` has removed the `no-div-regex` this is no longer as issue for me.

**Comment by platinumazure** [comment]
No worries, thanks for getting back to me so fast!

**Comment by SalsabeelQandeel** [comment]
I can work on this one, Is it available?

**Comment by platinumazure** [comment]
Hi @SalsabeelQandeel, feel free to work on this. Thanks!
